File: src/test/java/com/d20charactersheet/framework/boc/service/CharacterServiceTest.java
package com.d20charactersheet.framework.boc.service;

import org.junit.Before;

import com.d20charactersheet.framework.DnDv35Universe;
import com.d20charactersheet.framework.Universe;
import com.d20charactersheet.framework.dac.dao.dummy.storage.DnDv35CharacterStorage;

public class CharacterServiceTest extends BaseCharacterServiceTest {

  @Before
  public void setUp() {
    final Universe universe = new DnDv35Universe();
    gameSystem = universe.getGameSystem();
    final CharacterClassService characterClassService = gameSystem.getCharacterClassService();
    fighter = characterClassService.findClassByName("Fighter", gameSystem.getAllCharacterClasses());
    wizard = characterClassService.findClassByName("Wizard", gameSystem.getAllCharacterClasses());

    final RaceService raceService = gameSystem.getRaceService();
    human = raceService.findRaceByName("Human", gameSystem.getAllRaces());
  }

  @Override
  public void testDeleteCharacter() {
    // can only be implemented with an ImageService
  }

  @Override
  protected int getNumberOfCharacters() {
    return DnDv35CharacterStorage.CHARACTER.length;
  }
}

File: src/test/java/com/d20charactersheet/framework/boc/service/CharacterServiceTest.kt
package com.d20charactersheet.framework.boc.service

import com.d20charactersheet.framework.DnDv35Universe
import com.d20charactersheet.framework.Universe
import org.junit.Before

class CharacterServiceTest : BaseCharacterServiceTest() {

    @Before
    fun setUp() {
        val universe: Universe = DnDv35Universe()
        gameSystem = universe.gameSystem
        val characterClassService = gameSystem.characterClassService
        fighter = characterClassService.findClassByName("Fighter", gameSystem.allCharacterClasses)
        wizard = characterClassService.findClassByName("Wizard", gameSystem.allCharacterClasses)
        val raceService = gameSystem.raceService
        human = raceService.findRaceByName("Human", gameSystem.allRaces)
    }

    override fun testDeleteCharacter() {
        // can only be implemented with an ImageService
    }

    override fun getNumberOfCharacters(): Int {
        // number of characters in dndv35_phb_character.sql
        return 14
    }
}